specification/applicationinsights/resource-manager/Microsoft.Insights/stable/2015-05-01/myworkbooks_API.json

```diff
@@ -0,0 +1,554 @@
+        }
+      },
+      "description": "Workbook list result."
+    },
+    "MyWorkbook": {
+      "description": "An Application Insights private workbook definition.",
+      "allOf": [
+        {
+          "$ref": "#/definitions/MyWorkbookResource"
+        }
+      ],
+      "properties": {
+        "kind": {
+          "type": "string",
+          "description": "The kind of workbook. Choices are user and shared.",
+          "enum": [
+            "user",
+            "shared"
+          ],
+          "x-ms-enum": {
+            "name": "SharedTypeKind",
+            "modelAsString": true
+          }
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "description": "Metadata describing a workbook for an Azure resource.",
+          "$ref": "#/definitions/MyWorkbookProperties"
+        }
+      }
+    },
+    "MyWorkbookProperties": {
+      "description": "Properties that contain a private workbook.",
+      "required": [
+        "displayName",
+        "kind",
+        "category",
+        "serializedData",
+        "userId"
+      ],
+      "properties": {
+        "displayName": {
+          "type": "string",
+          "description": "The user-defined name of the private workbook."
+        },
+        "serializedData": {
+          "type": "string",
+          "description": "Configuration of this particular private workbook. Configuration data is a string containing valid JSON"
+        },
+        "version": {
+          "type": "string",
+          "description": "This instance's version of the data model. This can change as new features are added that can be marked private workbook."
+        },
+        "kind": {
+          "description": "Enum indicating if this workbook definition is owned by a specific user or is shared between all users with access to the Application Insights component.",
+          "x-ms-client-name": "SharedTypeKind",
+          "default": "shared",
+          "type": "string",
+          "enum": [
+            "shared",
+            "user"
+          ],
+          "x-ms-enum": {
+            "name": "SharedTypeKind",
+            "modelAsString": true
+          }
+        },
+        "timeModified": {
+          "type": "string",
+          "description": "Date and time in UTC of the last modification that was made to this private workbook definition.",
+          "readOnly": true
+        },
+        "category": {
+          "type": "string",
+          "description": "Workbook category, as defined by the user at creation time."
+        },
+        "tags": {
+          "type": "array",
+          "items": {
+            "type": "string"
+          },
+          "description": "A list of 0 or more tags that are associated with this private workbook definition"
+        },
+        "userId": {
+          "type": "string",
+          "description": "Unique user id of the specific user that owns this private workbook."
```

Comment:
Is it arbitrary string?

Reply:
it is read only and assigned by the service when created.  it is _possibly_ the aad object id for the user?

Reply:
If it's readonly, could we add it?

Reply:
kindof?  it is treated as readonly by the service, but if i tag it read only here am i going to have to touch all of the other examples/etc?  it was written in a way that it is returned in responses, and those responses can be updated and sent back in updates and the field will effectively ignored by the service, so you don't have to *remove* it to do a patch/etc.

>  code: 'READONLY_PROPERTY_NOT_ALLOWED_IN_REQUEST',
>                  params: [ 'userId', 'userId' ],
>                 message: 'ReadOnly property `"userId": "userId"`, cannot be sent in the request.',

i'm not sure there's a way to document system generated properties that are effectively "ignored" in a write?
i can mark it read only, but then have to delete it from the examples.